servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultMultiAddressUrlHttpClientBuilder.java

```diff
@@ -253,7 +256,13 @@ public StreamingHttpClient apply(final UrlKey urlKey) {
                 buildContext.builder.unresolvedAddressToHost(hostHeaderTransformer);
             }
 
-            if (HTTPS_SCHEME.equalsIgnoreCase(urlKey.scheme)) {
+            if (clientFilterFactory != null) {
+                buildContext.builder.appendClientFilter(clientFilterFactory.asClientFilter(urlKey.hostAndPort));
+            }
+
+            if (singleAddressConfigurator != null) {
+                singleAddressConfigurator.configure(urlKey.scheme, urlKey.hostAndPort, buildContext.builder);
+            } else if (HTTPS_SCHEME.equalsIgnoreCase(urlKey.scheme)) {
```

Comment:
The default behavior hasn't changed. Folks that make no code changes should see no behavior changes (e.g.  `secure()` works the same way it does today). If folks start using the new API, they will need to transition over to it (e.g. no half-in / half-out).

The `configurator` API is similar to `SchemeRegistry` AHC uses to register socket factories. We don't have an explicit `Scheme` type but the usage would be somewhat awkward here because `HostAndPort` is the second argument.

```java
SchemeRegistry registry = new SchemeRegistry();
registry.register(new Scheme("http", PlainSocketFactory.getSocketFactory(), 80));
registry.register(new Scheme("https", SSLSocketFactory.getSocketFactory(), 443));
```

The JDK client allows for sending requests to different URLs with the same client, but AFAICT it doesn't allow configuring different trust managers per request. The "one single default setting" works well for "browser" use cases but doesn't work cross trust domains in service to service use cases. In these cases you would have to create an independent client, which is what multi-address is doing under the hood (we just allow them to configure all the options).

```java
HttpClient c = HttpClient.newBuilder().sslContext(/*SSLContext*/).build();
HttpRequest req = HttpRequest.newBuilder(URI.create("https://www.servicetalk.io")).GET().build();
HttpResponse<T> response = c.send(req, /*response body handler*/);
```

Lets sync up offline to resolve this.

Reply:
Most users don't touch the `secure()` method at all, because they are satisfied with the default ssl config we provide for `https` scheme. With new API, their default behavior should remain the same, they should not care about invoking `sslConfig` now. This use-case should continue to work:
```java
BlockingHttpClient client = HttpClients.forMultiAddressUrl().buildBlocking();
client.request(client.get("https://servicetalk.io"));
```
Users who care about custom configuration, can use configurator to customize it.

Reply:
ah ok, I see what you are saying. thought the primary concern was related to passing the `scheme` in the configurator. I will refactor to preserve the "default trust manager always applied for https schema"